File: 118-PascalsTriangle/118-PascalsTriangle.cpp
// Last updated: 6/11/2026, 11:36:26 AM
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> pascal(n+1, vector<int>(n, 0));
        vector<vector<int>> ans;
        for(int i=1; i<=n; i++){
            vector<int> temp;
            for(int j=0; j<i; j++){
                if(i == 1){
                    pascal[i][j] = 1;
                }else if(j == 0 || j == i-1){
                    pascal[i][j] = 1;
                }else{
                    pascal[i][j] = pascal[i-1][j] + pascal[i-1][j-1];
                }
                temp.push_back(pascal[i][j]);
            }
            ans.push_back(temp);
        }
        return ans;
    }
};

File: 118-PascalsTriangle/118-PascalsTriangle.py
# Last updated: 6/11/2026, 11:36:34 AM
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        fans = []
        
        def fun(r):
            rowans= []
            ans = 1
            rowans.append(ans)
            for i in range(1,r):
                ans = ans*(r-i)
                ans = ans//(i)
                rowans.append(ans)
            k = rowans.copy()
            fans.append(k)

        for i in range(1,numRows+1):
            fun(i)
        return fans